engine/class_modules/priest/sc_priest.hpp

```diff
@@ -983,30 +982,48 @@ struct fiend_melee_t : public priest_pet_melee_t
   }
 };
 
+// ==========================================================================
+// Shadowflame Rift
+// ==========================================================================
+struct shadowflame_rift_t final : public priest_pet_spell_t
+{
+  shadowflame_rift_t( base_fiend_pet_t& p ) : priest_pet_spell_t( "shadowflame_rift", &p, p.o().find_spell( 344748 ) )
+  {
+    background = true;
+  }
+};
+
 // ==========================================================================
 // Shadowflame Prism
 // ==========================================================================
 struct shadowflame_prism_t final : public priest_pet_spell_t
 {
-  double shadowflame_increase;
+  double duration;
 
   shadowflame_prism_t( base_fiend_pet_t& p )
-    : priest_pet_spell_t( "shadowflame_prism", &p, p.o().find_spell( 336142 ) ),
-      shadowflame_increase( p.o().find_spell( 336144 )->effectN( 1 ).percent() )
+    : priest_pet_spell_t( "shadowflame_prism", &p, p.o().find_spell( 336143 ) ),
+      duration( as<double>( data().effectN( 3 ).base_value() ) + 0.5 )  // This 0.5 is hardcoded in spell data
   {
     background = true;
-    may_crit   = false;
-    may_miss   = false;
+
+    impact_action = new shadowflame_rift_t( p );
+    add_child( impact_action );
   }
 
-  void trigger( player_t* target, double original_amount )
+  void execute() override
   {
-    base_dd_min = base_dd_max = ( original_amount * shadowflame_increase );
-    player->sim->print_debug( "Triggered shadowflame prism damage on target {} at {} percent increase.", *target,
-                              shadowflame_increase );
+    priest_pet_spell_t::execute();
 
-    set_target( target );
-    execute();
+    auto current_pet = p().o().talents.mindbender->ok() ? p().o().pets.mindbender : p().o().pets.shadowfiend;
```

Comment:
You should be able to use the get_current_main_pet() function here. Also check if the current_pet variable you retrieve is null or not.